internal/hcsoci/hcsdoc_wcow.go

```diff
@@ -165,9 +165,20 @@ func createWindowsContainerDocument(ctx context.Context, coi *createOptionsInter
 		v2Container.Networking.NetworkSharedContainerName = v1.NetworkSharedContainerName
 	}
 
-	//	// TODO V2 Credentials not in the schema yet.
 	if cs, ok := coi.Spec.Windows.CredentialSpec.(string); ok {
 		v1.Credentials = cs
+		if coi.HostingSystem != nil {
+			return nil, nil, fmt.Errorf("v2 hyper-v isolated containers do not support gmsa")
+		}
+		// Hardcode false for uvm parameter as hyper-v not supported as of now.
+		// Later on have var xenon assigned to false above and assign true if
+		// coi.HostingSystem != nil
+		ccgState, err := CreateCredentialGuard(ctx, coi.actualID, cs, false)
```

Comment:
In general we don't do any real "work" in `createWindowsContainerDocument`, we should build up the document based on the passed in options. I think we should instead create the CCG instance in `resources_wcow.go`. To pass the CCG state back so it can be used when creating the HCS document, I suggest we add it to `createOptionsInternal`.

Reply:
Yea this was the main thing I didn't like about this approach. I like the createOptionsInternal idea :)